File: laboratório/LAB01Q02/MaiusculaRec.java
package LAB01Q02;

import java.util.*;

public class MaiusculaRec {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        while(sc.hasNextLine()){
            String texto = sc.nextLine();
            if(!(texto.equals("FIM")) && !(texto.isEmpty())){
                System.out.println(confereMaiuscula(texto, 0));
            }
        }
        sc.close();
    }

    public static int confereMaiuscula(String texto, int index){
        if (index >= texto.length()) {
            return 0; // Caso base: fim da string
        }
        int count = (texto.charAt(index) >= 'A' && texto.charAt(index) <= 'Z') ? 1 : 0;
        return count + confereMaiuscula(texto, index + 1); // Chamada recursiva
    }
}

File: laboratório/LAB01Q04/maiusculaRec.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int confereMaiuscula(const char *texto, int index);

int main() {
    char texto[100];

    while (fgets(texto, sizeof(texto), stdin)) {
        texto[strcspn(texto, "\n")] = '\0';

        if (strcmp(texto, "FIM") != 0 && strlen(texto) > 0) {
            printf("%d\n", confereMaiuscula(texto, 0));
        }
    }

    return 0;
}

int confereMaiuscula(const char *texto, int index) {
    if (index >= strlen(texto)) {
        return 0; // Caso base: fim da string
    }
    int count = isupper(texto[index]) ? 1 : 0;
    return count + confereMaiuscula(texto, index + 1);
}